### PR TITLE
Add isImageAvailable method to Metadata

### PR DESCRIPTION
**Pull request by jo66**
## Subject

This PR will allow use of default mosaic background (see [admin bundle PR #5366](https://github.com/sonata-project/SonataAdminBundle/pull/5366))

I am targeting this branch, because it's feature.

## Changelog

```markdown
### Added
- Added `isImageAvailable` method to `Metadata` class to allow use of default mosaic background
```

## To do

- [x] Update the tests

### CODE REVIEW

File: src/CoreBundle/Model/MetadataInterface.php [hunk 1]
```diff
@@ -33,6 +33,11 @@ public function getDescription();
      */
     public function getImage();
 
+    /**
+     * @return boolean
+     */
+    public function isImageAvailable();
```
Comment (kunicmarko20) [inline src/CoreBundle/Model/MetadataInterface.php]: bc break

Comment (jo66) [reply src/CoreBundle/Model/MetadataInterface.php]: Removed change on interface

File: src/CoreBundle/Model/Metadata.php [hunk 2]
```diff
@@ -60,6 +65,7 @@ public function __construct($title, $description = null, $image = null, $domain
...
+        $this->isImageAvailable = null !== $image;
```
Comment (greg0ire) [inline src/CoreBundle/Model/Metadata.php]: No need to store this in a field IMO: computing it every time should be performant enough

Comment (jo66) [reply src/CoreBundle/Model/Metadata.php]: Yes, but as image will be initialized with default value, how to know if it's default or not after initialization? To check image value against default one will be less performant I guess...

File: src/CoreBundle/Model/Metadata.php [hunk 3]
```diff
@@ -39,6 +39,11 @@ class Metadata implements MetadataInterface
      */
     protected $image;
 
+    /**
+     * @var boolean
+     */
+    private $isImageAvailable;
```
Comment (core23) [inline src/CoreBundle/Model/Metadata.php]: This class is deprecated, we shouldn't add any features to this class anymore

Comment (jo66) [reply src/CoreBundle/Model/Metadata.php]: In this case admins that still use it won't benefit from mosaic background... Not a problem? Should I add a deprecation notice about it or just close PR?

File: src/CoreBundle/Model/Metadata.php (hunk 2 above)
Comment (greg0ire) [reply src/CoreBundle/Model/Metadata.php]: You can store the default value in a constant and do the comparison against it, performance will really not be an issue here IMO.

Comment (jo66) [reply src/CoreBundle/Model/Metadata.php]: Done

File: tests/CoreBundle/Model/MetadataTest.php [hunk 4]
```diff
@@ -48,12 +48,22 @@ public function testGetters()
...
     {
         $metadata = new Metadata('title', 'description', false);
         $this->assertFalse($metadata->getImage());
     }
+
+    public function testIsImageAvailable()
+    {
+        $metadata = new Metadata('title', 'description');
+        $this->assertFalse($metadata->isImageAvailable());
+        $metadata = new Metadata('title', 'description', false);
```
Comment (OskarStark) [inline tests/CoreBundle/Model/MetadataTest.php]: Pleas use a dataProvider instead

Comment (jo66) [reply tests/CoreBundle/Model/MetadataTest.php]: Done.

**Comment by core23** [comment]
We've added a new `Metadata` object to the AdminBundle. So we could close this one, as it is deprecated and have no special domain context.

**Comment by jo66** [comment]
@core23 Thanks for this, I can now use it for default background in admin bundle :)

**Pull request closed without merging**